### PR TITLE
Callback for exitCode

### PR DESCRIPTION
**Pull request by sarelvdwalt**
Addresses #11 to create a callback in userland for exitCodes returned by processes.

### CODE REVIEW

**Comment by sarelvdwalt** [comment]
The method chosen to do this was to add a configuration function `setCompletedCallback`. The way of using this is then:

```php
$swarmProcess = new SwarmProcess($logger);
$swarmProcess->setCompletedCallback(function(Symfony\Component\Process\Process $process) {
    // do something with the $process returned, checking it's exit code, and perhaps putting it back on the stack
});
```

**Comment by sarelvdwalt** [comment]
@mostertb and @johnathanmdell would you mind having a look and telling me what your thoughts are?

**Comment by johnathanmdell** [comment]
@sarelvdwalt I'm happy with that, nothing overly complicated to achieve some simple communication between the child and parent.

👍

**Comment by mostertb** [comment]
@sarelvdwalt This looks great! 
Its something that I've personally very much wanted. Thank you

Two things:
1) Will you update the `README.md` with this new functionality?
2) If running in a loop with `$swarm->tick()`, the callback could theoretically be changed mid-execution. Is this something we intend to support?

**Comment by sarelvdwalt** [comment]
> @sarelvdwalt This looks great!
> Its something that I've personally very much wanted. Thank you
> 
> Two things:
> 
> 1. Will you update the `README.md` with this new functionality?
> 2. If running in a loop with `$swarm->tick()`, the callback could theoretically be changed mid-execution. Is this something we intend to support?

Thanks @mostertb 
1. Ok will do and notify
2. Not sure, didn't think about this - I see no reason why not to allow this, other than protecting the user from themselves. I think we should leave this open.

**Comment by sarelvdwalt** [comment]
@mostertb I have updated the documentation.

File: src/Configuration.php [hunk 1]
```diff
@@ -0,0 +1,32 @@
+<?php
+namespace Afrihost\SwarmProcess;
+
+class Configuration
+{
+    /** @var callable */
```
Comment (sarelvdwalt) [inline src/Configuration.php:6]: fix comment

**Comment by sarelvdwalt** [comment]
@mostertb and @johnathanmdell I have now moved the callback to the Configuration object. This object creation is also to plan for the future change planned for a setting to respect Process timeout settings.

**Review by mostertb — APPROVED** [review]

**Pull request merged**